src/yunohost/certificate.py

```diff
@@ -596,10 +596,9 @@ def _prepare_certificate_signing_request(domain, key_file, output_folder):
     # For "parent" domains, include xmpp-upload subdomain in subject alternate names
     if domain in domain_list(exclude_subdomains=True)["domains"]:
         subdomain = "xmpp-upload." + domain
-        try:
-            _dns_ip_match_public_ip(get_public_ip(), subdomain)
+        if dig(subdomain, "A", resolvers="force_external") == ("ok", [get_public_ip()]):
```

Comment:
And what about IPv6 ?

Reply:
Hmyea indeed, it may be not so hard to fetch the result from the diagnosis system as well, will have a look